### ISSUE TITLE
File option type doesn't work in native

### ISSUE THREAD

**Issue by wirthi** [spring-projects/spring-shell#960]
Hi,

**tl;dr:** I believe you are missing native reflection configuration for at least `ch.qos.logback.classic.joran.SerializedModelConfigurator` for a Spring Shell app to properly work with the native compile feature.

I have used Spring Initializr to create a Spring Boot 3.2.0 app and included Spring Shell as (only) dependency. Java 21, Maven.

[demo.zip](https://github.com/spring-projects/spring-shell/files/13744406/demo.zip)
[demoWithNativeBuildTools.zip](https://github.com/spring-projects/spring-shell/files/13744466/demoWithNativeBuildTools.zip) (edit: have added this file that includes the proper config in pom.xml)

When the native compile feature is used (GraalVM) on the generated application (without modifications, except adding native-build-tools configuration), an executable is successfully generated. That executable starts, but fails with a NullPointerException (in the constructor of FileInputStream) on e.g. this command:
```
shell --file AValidFileName
```
The NPE is not of relevance, it is a consequence of the actual problem: **there is incomplete configuration for the reflection at runtime**. Using the `-H:ThrowMissingRegistrationErrors=true` option of GraalVM native-image (23.1 or later) I see the following problem (note: this is print when executing the generated image):
`
Exception in thread "main" org.graalvm.nativeimage.MissingReflectionRegistrationError: The program tried to reflectively access class ch.qos.logback.classic.joran.SerializedModelConfigurator without it being registered for runtime reflection. Add ch.qos.logback.classic.joran.SerializedModelConfigurator to the reflection metadata to solve this problem. See https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection for help.
`
This message typically means that this dependency is using reflection, but you don't provide reflection configuration, so native-image generates an image that can only fail when this codepath is taken - it throws the NullPointerException.

Note:
- I did add the `native-build-tools` plugin and metadataRepository config as described [in your documentation](https://docs.spring.io/spring-shell/docs/3.1.6/docs/#native). This is NOT in the attached file however.
- `./mvnw package -Pnative` as listed in the documentation does not trigger a native compile for me. I am using `./mvnw -Pnative native:compile` instead.

(discussed in https://graalvm.slack.com/archives/GPEQFN084/p1703162180614929)

Thanks,
Christian Wirth
GraalVM Native Image

**Comment by jvalkeal** [comment]
That slack link(is it on channel I don't have access?) is not working for me so I can't see what is the discussion. Looks like actual code giving some trouble is not part of the zips you included.

**Comment by sdeleuze** [comment]
It is on GraalVM Slack, I will share with you the related discussion.

**Comment by jvalkeal** [comment]
I got more details from a slack thread.

**Comment by jvalkeal** [comment]
I was looking the root cause of this which seems to be a failure of spring's `ConversionServicefrom` String to a File. spring-shell currently relies on `FormattingConversionService` and goes to `ObjectToObjectConverter` and creates File based on a given option value. This doesn't seem to work out-of-the-box with spring-boot with graal.

If I replace to use boot's `ApplicationConversionService` which registers its own `StringToFileConverter` this String to File conversion starts to work just fine under graal.

Essentially under graal:
```
DefaultConversionService cs = new DefaultConversionService();
// canConvert is false
boolean canConvert = cs.canConvert(String.class, File.class);
```
